src/http.c

```diff
@@ -471,6 +484,28 @@ http_request_free(struct http_request *req)
 		kore_pool_put(&http_header_pool, hdr);
 	}
 
+	for (ck = TAILQ_FIRST(&(req->resp_cookies)); ck != NULL; ck = cknext) {
+		cknext = TAILQ_NEXT(ck, list);
+
+		TAILQ_REMOVE(&(req->resp_cookies), ck, list);
+		kore_free(ck->name);
+		kore_free(ck->value);
+		if (ck->path != NULL)
```

Comment:
kore_free() knows how to deal with NULL pointers so no need to check for NULL before calling it.